scripts/core/get-superdesk-api-implementation.ts

```diff
@@ -0,0 +1,49 @@
+import {ISuperdesk, IExtensions} from 'superdesk-api';
+import {gettext} from 'core/utils';
+
+export function getSuperdeskApiImplementation(
+    requestingExtensionId: string,
+    extensions: IExtensions,
+    modal,
+): ISuperdesk {
+    return {
+        ui: {
+            alert: (message: string) => modal.alert({bodyText: message}),
+            confirm: (message: string) => new Promise((resolve) => {
+                modal.confirm(message, gettext('Cancel'))
+                    .then(() => resolve(true))
+                    .catch(() => resolve(false));
+            }),
+        },
+        localization: {
+            gettext: (message) => gettext(message),
+        },
+        extensions: {
+            getExtension: (id: string) => {
+                const extension = extensions[id].extension;
+
+                if (extension == null) {
+                    console.error('Extension not found.');
+                    return Promise.resolve(null);
+                }
+
+                const {manifest} = extensions[requestingExtensionId];
+
+                if (
+                    manifest.superdeskExtension != null
+                    && Array.isArray(manifest.superdeskExtension.dependencies)
+                    && manifest.superdeskExtension.dependencies.includes(id)
+                ) {
+                    const extensionShallowCopy = {...extension};
+
+                    delete extensionShallowCopy['activate'];
+
+                    return Promise.resolve(extensionShallowCopy);
+                } else {
+                    console.error('Not authorized.');
+                    return Promise.resolve(null);
```

Comment:
what about `Promise.reject` here? so it would not call any `.then` code if extension is not available

Reply:
Good catch. I guess I forgot I can reject promises.